src/Assignments/ProblemIframe.tsx

```diff
@@ -71,6 +79,79 @@ export const ProblemIframe: React.FC<ProblemIframeProps> = ({
         setHeight(`${scrollHeight}px`);
     };
 
+    const formDataToObject = (formData: FormData) => {
+        let object:any = {};
+        // @ts-ignore
+        for(let pair of formData.entries()) {
+            if (_.isUndefined(object[pair[0]])) {
+                object[pair[0]] = pair[1];
+            } else {
+                if(!_.isArray(object[pair[0]])) {
+                    object[pair[0]] = [object[pair[0]]];
+                }
+                object[pair[0]].push(pair[1]);
+            }
+        }
+        return object;
+    };
+
+    async function prepareAndSubmit(problemForm: HTMLFormElement, clickedButton?: HTMLButtonElement) {
+        const submitAction = (window as any).submitAction;
+        if(typeof submitAction === 'function') submitAction(); // this is a global function from renderer - prepares form field for submit
+
+        let formData = new FormData(problemForm);
+        if (_.isNil(problem.grades)) {return;} // just for typescript
+        if (_.isNil(problem.grades[0])) {return;} // not enrolled
+        if (_.isNil(problem.grades[0].id)) {
+            // PANIC -- should not happen
+            setError(`No grades id for problem #${problem.id}`);
```

Comment:
Maybe we should do a console.error here as well or instead

Once we get the remote logging in I plan to do a global search for console.error to find where we need it

I'm thinking the remote logger will return a reference number and we can set the error to:
`Unexpected application error, contact rederly support with Rederly Reference Number #EXAMPLE_NUMBER` (since there is no corrective action for the user here)